crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -275,26 +274,19 @@ where
             .iter()
             .filter(|key| is_masp_tx_pin_key(key))
             .collect();
-        match pin_keys.len() {
-            0 => (),
-            1 => {
-                match self
-                    .ctx
-                    .read_post::<IndexedTx>(pin_keys.first().unwrap())?
-                {
-                    Some(IndexedTx { height, index, .. })
-                        if height == self.ctx.get_block_height()?
-                            && index == self.ctx.get_tx_index()? => {}
-                    Some(_) => {
-                        return Err(Error::NativeVpError(
-                            native_vp::Error::SimpleMessage(
-                                "Invalid MASP pin key",
-                            ),
-                        ));
-                    }
-                    _ => (),
+        match &pin_keys[..] {
+            [] => (),
+            [pin_key] => match self.ctx.read_post::<IndexedTx>(pin_key)? {
+                Some(IndexedTx { height, index, .. })
+                    if height == self.ctx.get_block_height()?
+                        && index == self.ctx.get_tx_index()? => {}
+                Some(_) => {
+                    return Err(Error::NativeVpError(
+                        native_vp::Error::SimpleMessage("Invalid MASP pin key"),
+                    ));
                 }
-            }
+                None => (),
+            },
```

Comment:
move to `validate_pin_key` helper method

Reply:
Thanks. In #3264, this is no longer an issue because pinned key support has been removed.